### DISCUSSION TITLE
Downloading a zip file from the endpoint is returning 403 no matter what

### DISCUSSION THREAD

**Discussion by mariuspetrutillinger** [community/community#81447] [Apps, API and Webhooks]
### Select Topic Area

Question

### Body

Hello everyone. I am currently attempting to download a zip file of a repository master branch using the REST API but no matter what I do it doesn't seem to work. I tried 2 different URI: https://api.github.com/{owner}/{repo}/archive/{branch}.zip and https://api.github.com/repos/{owner}/{repo}/zipball/{branch}. I used a personal access token every single time and it is a functional access token, however nothing seems to happen. 

Any pointers regarding what I should do or what could be the reason why this doesn't work?

**Comment by bydevmar** [comment]
Hello! It seems you're facing issues with downloading a zip file from a GitHub repository using the REST API. A couple of things to check:

1. **Authentication:** Ensure that your personal access token has the necessary permissions to access the repository. It should have at least the "repo" scope.

2. **Correct Endpoint:** The correct endpoint for downloading a zip file of the master branch is `https://codeload.github.com/{owner}/{repo}/legacy.zip/{branch}`. Make sure you are using this format.

Here's an example using curl:

```bash
curl -LJO https://codeload.github.com/{owner}/{repo}/legacy.zip/{branch} -H "Authorization: token YOUR_ACCESS_TOKEN"
```

Replace `{owner}`, `{repo}`, `{branch}`, and `YOUR_ACCESS_TOKEN` with your actual values.

If the issue persists, consider checking the GitHub API documentation for any updates or changes to the endpoints. Additionally, reviewing the API response for error messages might provide insights into what's going wrong.

**Reply by mariuspetrutillinger** [reply]
I think the token is alright and functional. When I try to use the endpoint you mentioned, it gives me a 404. I couldn't find anything about it in the api documentation. Could you point me to where I should look?

**Reply by mariuspetrutillinger** [reply]
When I say I didn't find anything about it I meant I couldn't find anything apart from what I said in the top post which is this 
![image](https://github.com/community/community/assets/74731592/3183c92d-2e7e-4fe6-a1cf-9418ae6c17ea)
